website/server/controllers/api-v3/tasks/groups.js

```diff
@@ -200,13 +200,25 @@ api.assignTask = {
     if (canNotEditTasks(group, user, assignedUserId)) throw new NotAuthorized(res.t('onlyGroupLeaderCanEditTasks'));
 
     let promises = [];
-
-    if (user._id !== assignedUserId) {
-      const taskText = task.text;
-      const managerName = user.profile.name;
-
+    const taskText = task.text;
+    const userName = user.profile.name;
+
+    if (user._id === assignedUserId) {
+      const managerIds = Object.keys(group.managers);
+      managerIds.push(group.leader);
+      const managers = await User.find({_id: managerIds}, 'notifications').exec();
```

Comment:
instead of loading all the managers and then saving them which is quite slow, can we just push the notification with a direct update? using there's `pushNotification` under the User model's static methods

Reply:
ohh, I thought that method was about _push_ notifications specifically, like for mobile devices 🤦‍♂ I'll try that!

Reply:
Yeah that's a bit confusing